scripts/utils.js

```diff
@@ -0,0 +1,53 @@
+const hre = require('hardhat');
+
+const networks = {
+  mainnets: [
+    'mainnet',
+    'arbitrum',
+    'avalanche',
+    'bsc',
+    'fantom',
+    'gnosis',
+    'metis',
+    'milkomeda',
+    'moonbeam',
+    'moonriver',
+    'optimism',
+    'polygon',
+    'rsk',
+  ],
+  testnets: [
+    'goerli',
+    'sepolia',
+    'avalanche_testnet',
+    'milkomeda_testnet',
+    'fantom_testnet',
+    'moonbase_testnet',
+    'bsc_testnet',
+    'rsk_testnet',
+    'polygon_testnet',
+    'arbitrum_testnet',
+    'optimism_testnet',
+    'gnosis_testnet',
+  ],
+};
+
+const apis = {
+  mainnets: 'ANU Quantum Random Numbers',
+  testnets: 'BYOG',
+};
+
+function detectApi(network) {
+  const netType = Object.keys(networks).filter((net) => {
+    return networks[net].includes(network.name);
```

Comment:
Why would you use `includes()`, are you not looking for an exact match?

Reply:
If anything, this will return both the mainnet and testnet when you filter with the mainnet name.

Reply:
`includes()` should return false in such case where array is ['something-testnet'] and you are looking for `something`.

Reply:
Alright, I thought this was something like `'something-testnet'.includes('something')`